src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -5057,7 +5057,10 @@ QVariant QgsPostgresProvider::convertValue( QVariant::Type type, QVariant::Type
         result = QVariant( type );
       break;
     case QVariant::UserType:
-      result = fromEwkt( value, conn );
+      if ( typeName == QLatin1String( "geometry" ) )
+        result = fromEwkt( value, conn );
+      if ( typeName == QLatin1String( "interval" ) )
+        result = QgsInterval::fromString( value );
```

Comment:
```suggestion
      else if ( typeName == QLatin1String( "interval" ) )
        result = QgsInterval::fromString( value );
      else
        qCritical() << "Unknown typeName " << typeName;
```

Maybe you can add even more context